source/kraken/maintenance_worker.cpp

```diff
@@ -56,14 +56,130 @@ namespace bg = boost::gregorian;
 
 namespace navitia {
 
+void MaintenanceWorker::run() {
+    // try to load data from disk
+    load_data();
+
+    if (!is_data_loaded()) {
+        // The data is not loaded, so we wait for a reload message on the task queue
+        // that results in a successfull data load.
+        // We may be disconnected from rabbitmq during this wait,
+        // and reconnection to rabbitmq is handled by this while(true) loop
+        while (true) {
+            try {
+                open_channel_to_rabbitmq();
+                create_task_queue();
+                // will return when a reload message on the task queue
+                // arrived and resulted in a successfull data load
+                listen_to_task_queue_until_data_loaded();
+                // data is loaded, let's break from the while(true)  loop
+                break;
+            } catch (const std::runtime_error& ex) {
+                LOG4CPLUS_ERROR(logger, "Connection to rabbitmq failed: " << ex.what());
+                data_manager.get_data()->is_connected_to_rabbitmq = false;
+                sleep(10);
+            }
+        }
+    } else {
+        // the data is loaded, so we just try to connect to rabbitmq and create
+        // the task queue
+        try {
+            open_channel_to_rabbitmq();
+            create_task_queue();
+        } catch (const std::runtime_error& ex) {
+            LOG4CPLUS_ERROR(logger, "Connection to rabbitmq failed: " << ex.what());
+            data_manager.get_data()->is_connected_to_rabbitmq = false;
+        }
+    }
+
+    // if we connected to rabbitmq above
+    // then the reload queue is already connected, and we just have
+    // to bind the realtime queue and begin listening to rabbitmq
+    // otherwise, rabbitmq is not connected and this block will fail immediatly
+    try {
+        create_realtime_queue();
+        listen_rabbitmq();
+    } catch (const std::runtime_error& ex) {
+        LOG4CPLUS_ERROR(logger, "Connection to rabbitmq failed: " << ex.what());
+        data_manager.get_data()->is_connected_to_rabbitmq = false;
+    }
+
+    // if rabbitmq was not connected, or was disconnected during the previous listen_rabbitmq()
+    // we try to reconnect to rabbitmq
+    // When reconnection is successfull, we don't want to load_data(), we just listen to the queues
+    while (true) {
+        try {
+            open_channel_to_rabbitmq();
+            create_task_queue();
+            create_realtime_queue();
+            listen_rabbitmq();
+        } catch (const std::runtime_error& ex) {
+            LOG4CPLUS_ERROR(logger, "Connection to rabbitmq failed: " << ex.what());
+            data_manager.get_data()->is_connected_to_rabbitmq = false;
+            sleep(10);
```

Comment:
(sorry to be a pest, I know that was not your code, it seems to be mine :see_no_evil: ) :
```suggestion
            std::this_thread::sleep_for(10);
```